src/fileparse.py

```diff
@@ -0,0 +1,103 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, Callable, List, TextIO, Union, Dict
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    convert_fn: Union[List[Callable], Dict[str, Callable]] = None,
```

Comment:
To be complete regarding the typing you should wrap the typing with `Optional` since it can be `None`:
```suggestion
    convert_fn: Optional[Union[List[Callable], Dict[str, Callable]]] = None,
```

Reply:
Added `Optional`.